File: dotnetv3/IAM/GetPolicyExample/GetPolicy.cs
// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
// SPDX-License-Identifier:  Apache-2.0

// This example shows how to get information about an AWS Identity and Access
// Management (IAM) policy using the Amazon Resource Name (ARN) of the policy.
// The example was created using the AWS SDK for .NET 3.7 and .NET Core 5.

// snippet-start:[IAM.dotnetv3.GetPolicyExample]

using System;
using Amazon.IdentityManagement;
using Amazon.IdentityManagement.Model;

var client = new AmazonIdentityManagementServiceClient();
var request = new GetPolicyRequest
{
    PolicyArn = "POLICY_ARN",
};

var response = await client.GetPolicyAsync(request);

Console.Write($"{response.Policy.PolicyName} was created on ");
Console.WriteLine($"{response.Policy.CreateDate}");

// snippet-end:[IAM.dotnetv3.GetPolicyExample]

File: javascriptv3/example_code/iam/actions/get-policy.js
/*
 * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
 * SPDX-License-Identifier: Apache-2.0
 */

import { fileURLToPath } from "url";

// snippet-start:[iam.JavaScript.policies.getPolicyV3]
import { GetPolicyCommand, IAMClient } from "@aws-sdk/client-iam";

const client = new IAMClient({});

/**
 *
 * @param {string} policyArn
 */
export const getPolicy = (policyArn) => {
  const command = new GetPolicyCommand({
    PolicyArn: policyArn,
  });

  return client.send(command);
};
// snippet-end:[iam.JavaScript.policies.getPolicyV3]

// Invoke main function if this file was run directly.
if (process.argv[1] === fileURLToPath(import.meta.url)) {
  getPolicy("POLICY_ARN");
}
